pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactorMXBeanImpl.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.compaction;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.LongAdder;
+
+public class CompactorMXBeanImpl implements CompactorMXBean {
+
+    private final ConcurrentHashMap<String, LongAdder> topicRemovedEventOps = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Long> lastCompactSucceedTimestampOps = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Long> lastCompactFailedTimestampOps = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<String, Long> lastCompactDurationTimeOps = new ConcurrentHashMap<>();
+
+    public void addCompactRemovedEvent(String topic) {
+        topicRemovedEventOps.computeIfAbsent(topic, k -> new LongAdder()).increment();
+    }
+
+    public void addCompactStartOp(String topic) {
+        topicRemovedEventOps.remove(topic);
+        lastCompactDurationTimeOps.put(topic, System.nanoTime());
```

Comment:
If the compact task is not complete, we will get a very large value here.

Reply:
I have fixed this issue by adding a new field to mark the start time in a later update.